hera_cal/reflections.py

```diff
@@ -51,169 +72,80 @@
+            edgecut_hi : int, Nbins to flag but not window at high-side of the FFT axis.
+            window : str, tapering function to apply across freq before FFT
             alpha : float, if taper is Tukey, this is its alpha parameter
-            zero_pad : int, number of channels to pad band edges with zeros before FFT
+            zeropad : int, number of channels to pad band edges with zeros before FFT
             fthin : int, scaling factor to down-select frequency axis when solving for phase
+            ref_sig_cut : float, if max reflection significance is not above this, do not record solution
             overwrite : bool, if True, overwrite dictionaries
+
+        Result:
+            self.ref_eps : dict, reflection epsilon term
+            self.ref_amp : dict, reflection amplitude
+            self.ref_phs : dict, reflection phase [radians]
+            self.ref_dly : dict, reflection delay [nanosec]
+            self.ref_flags : dict, reflection flags [bool]
+            self.ref_significance : dict, reflection significance,
+                which is max|Vfft| / median|Vfft| near reflection peak.
         """
         # initialize containers
         if hasattr(self, 'epsilon') and not overwrite:
             raise ValueError("reflection dictionaries exist but overwrite is False...")
-        self.epsilon = {}
-        self.amps = {}
-        self.phs = {}
-        self.delays = {}
-        self.peak_ratio = {}
-        self.dfft = DataContainer({})
-
-        # configure data
-        if data == 'clean':
-            data = self.clean_data
-            freqs = self.clean_freqs
-            self.reflection_times = self.clean_times
-        elif data == 'data':
-            # if using unCLEANed data, apply flags now on-the-fly
-            data = DataContainer(dict([(k, self.data[k] * ~self.flags[k]) for k in self.data.keys()]))
-            freqs = self.freqs
-            self.reflection_times = self.times
-        else:
-            raise ValueError("Didn't recognize data dictionary {}".format(data))
-
-        # get keys: only use auto correlations to model reflections
+        self.ref_eps = {}
+        self.ref_amp = {}
+        self.ref_phs = {}
+        self.ref_dly = {}
+        self.ref_flags = {}
+        self.ref_significance = {}
+
+        # get flags
+        if flags is None:
+            flags = DataContainer(dict([(k, np.zeros_like(data[k], dtype=np.bool)) for k in data]))
+
+        # get keys: only use auto correlations and auto pols to model reflections
         if keys is None:
-            keys = [k for k in data.keys() if k[0] == k[1]]
+            keys = [k for k in data.keys() if k[0] == k[1] and k[2][0] == k[2][1]]
 
-        # get window kwargs
-        win_kwargs = {}
-        if taper == 'tukey':
-            win_kwargs['alpha'] = alpha
+        # Take FFT of data
+        self.fft_data(data, flags=flags, keys=keys, assign='dfft', ax='freq', window=window, alpha=alpha,
```

Comment:
In general, I don't really get the philosophy here. Why assign the results to `self.dfft` rather than returning them and operating on them locally? Do we need them to stick around in the namespace and the memory?

Also, you could further reduce the memory footprint by doing the fft_data inside the following `for k in keys` loop and then letting each dfft[k] go out of scope after each iteration of the loop.

Reply:
In practice I've been wanting to use `dfft` for analysis (like when I make plots to compare before and after), and `self.fft_data` has some convenience to it over the `fft_data` function. The memory footprint concern is well-taken, but b/c this is running on autocorrs only I don't think its going to be an issue for us.